streams/src/main/java/org/apache/kafka/streams/kstream/Suppressed.java

```diff
@@ -155,6 +146,25 @@ static StrictBufferConfig unbounded() {
      * @return a suppression configuration
      */
     static <K> Suppressed<K> untilTimeLimit(final Duration timeToWaitForMoreEvents, final BufferConfig bufferConfig) {
-        return new SuppressedInternal<>(timeToWaitForMoreEvents, bufferConfig, null, false);
+        return new SuppressedInternal<>(null, timeToWaitForMoreEvents, bufferConfig, null, false);
     }
+
+    /**
+     * Use the specified name for the suppression node in the topology and for its changelog.
+     * <p>
+     * This can be used to insert a suppression without changing the rest of the topology names
+     * (and therefore not requiring a reset).
+     * <p>
+     * Note however, that once a suppression has buffered some records, removing it from the topology would cause
+     * the loss of those records.
+     * <p>
+     * A suppression can be "disabled" with the configuration {@code untilTimeLimit(Duration.ZERO, ...}.
+     * <p>
+     * An advanced option is available at this point: a "disabled" suppression whose metrics report that it is empty
```

Comment:
Do we want this in the JavaDocs? Or just AK docs?

Reply:
I'm not sure. I've removed the last sentence.